modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -688,11 +689,10 @@ std::vector<TensorInfo> ONNXCompiled::getTensorInfo(TensorPosition pos) {
             : this_session.GetOutputTypeInfo(i);
         tensor_info.emplace_back(info.GetTensorTypeAndShapeInfo());
 
-        char *name_p = pos == INPUT
-            ? this_session.GetInputName(i, allocator)
-            : this_session.GetOutputName(i, allocator);
-        tensor_info.back().name = name_p;
-        allocator.Free(name_p);
+        Ort::AllocatedStringPtr name_p = pos == INPUT
+            ? this_session.GetInputNameAllocated(i, allocator)
+            : this_session.GetOutputNameAllocated(i, allocator);
+        tensor_info.back().name = std::string(name_p.get());
```

Comment:
@dmatveev, please, check that copy is created properly, because AllocatedStringPtr name_p will die in this scope.

Try run demos or onnx tests. I did not see onnx check (mb I am blind).

Reply:
`Existing tests pass with the ONNX models mentioned in tests.`
Ok. Then all works.

Reply:
@dmatveev Did the G-API ONNX classification test pass on your end (requires model downloading)? The model I have keeps returning garbage and inference results don't differ at all while I process my video sample. The behavior is the same on the CPU and GPU.

Reply:
@mpashchenkov thanks for review, std::string copies its `char*` argument (not moves) so its ok for `name_p` to be destroyed right after.

@asutic I will have a look on that, thanks!

Reply:
@dmatveev,
A crutch for the longevity of names:
Keep ptrs in class state ->
```
std::vector<const char*> inputNodeNames;
std::vector<Ort::AllocatedStringPtr> inputNodePtrs;
```
just in case *